pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -20,14 +21,25 @@ import (
 	"github.com/rancher/types/config"
 	"github.com/rancher/types/user"
 	"github.com/sirupsen/logrus"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/runtime"
+	k8stypes "k8s.io/apimachinery/pkg/types"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
 const AgentForceDeployAnn = "io.cattle.agent.force.deploy"
 
+var (
+	nodeAgentImages    map[k8stypes.UID]string
```

Comment:
This should be one map and one lock, the key should be cluster.ID, this will be much easier to debug if needed vs a UID. 
Use a type `map[string]map[string]string` > clusterID['nodeAgentImage']foo